### PR TITLE
Support Local Testing Of Terraform

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
Need to run integration testing inside personal aws accounts

# Description of changes
Add extra var and readme on how to test in personal aws accounts 

# Tests
Followed the readme

### CODE REVIEW

File: integration/terraform/ec2/linux/README.md [hunk 1]
```diff
@@ -19,3 +19,64 @@ AMI builder pipeline builds the ami
...
+         1. You may want to do this on a linux ec2 instance installing the agent may fail if you build on Mac 
+         2. If you want to build faster remove not needed packages and test from release step
+   2. aws s3 cp build/bin s3://${your bucket name}/integration-test/binary/${git commit sha} --recursive
+3. Start Local Stack
+   1. Go to Local Stack directory
+      1. cd ../localstack
+   2. init terraform
+      1. terraform init
+   3. Apply terraform 
+      1. ````
```
Comment (SaxyPandaBear) [inline integration/terraform/ec2/linux/README.md:42]: Nitpick: there's an extra ` char for enclosing this code block

File: integration/terraform/ec2/linux/main.tf [hunk 2]
```diff
@@ -36,7 +36,7 @@ resource "aws_instance" "integration-test" {
 
 data "aws_ami" "latest" {
   most_recent = true
-  owners      = ["self"]
+  owners      = ["self", "506463145083"]
```
Comment (SaxyPandaBear) [inline integration/terraform/ec2/linux/main.tf:39]: Where's this new owner string coming from?

File: integration/terraform/ec2/linux/README.md [hunk 3]
```diff
@@ -19,3 +19,64 @@ AMI builder pipeline builds the ami
 The pipeline installs required packages and updates ami software
 
 This process generates a new ami we can then use for testing
+
+**How to integration test in your aws account**
+1. Create resources and setup local
+   1. Install terraform
```
Comment (SaxyPandaBear) [inline integration/terraform/ec2/linux/README.md:25]: Should there be a step for installing localstack?

Comment (sethAmazon) [reply integration/terraform/ec2/linux/README.md:25]: No. Local stack run on an ec2 instance. The instance ami has docker already installed.

File: integration/terraform/ec2/linux/main.tf (hunk 2 above)
Comment (sethAmazon) [reply integration/terraform/ec2/linux/main.tf:39]: This is the cw agent open source test account. You need this to be able to find the ami I made public from the image builder pipeline.

File: integration/terraform/ec2/linux/README.md [hunk 4]
```diff
@@ -19,3 +19,64 @@ AMI builder pipeline builds the ami
...
+   2. Set up aws credentials
+   3. Set up iam role
+      1. Role must include cloud watch server policy and s3 policy
+   4. Create s3 bucket
+2. Upload binary to s3
+   1. make release 
+      1. hints
+         1. You may want to do this on a linux ec2 instance installing the agent may fail if you build on Mac 
+         2. If you want to build faster remove not needed packages and test from release step
```
Comment (adam-mateen) [inline integration/terraform/ec2/linux/README.md:34]: Could change this to recommend users run:
`make build && make package-rpm` or `package-deb` (depending on which OS type will be tested).

File: integration/terraform/ec2/linux/README.md [hunk 5]
```diff
@@ -19,3 +19,64 @@ AMI builder pipeline builds the ami
...
+   2. aws s3 cp build/bin s3://${your bucket name}/integration-test/binary/${git commit sha} --recursive
+3. Start Local Stack
+   1. Go to Local Stack directory
+      1. cd ../localstack
```
Comment (adam-mateen) [inline integration/terraform/ec2/linux/README.md:38]: I recommend changing all the relative path to include the path from the git repository root directory, for example:
`cd ../localstack` to `cd `amazon-cloudwatch-agent/integration/terraform/ec2/localstack/`.

File: integration/terraform/ec2/linux/README.md [hunk 6]
```diff
@@ -19,3 +19,64 @@ AMI builder pipeline builds the ami
...
+      1. ```
+         terraform apply --auto-approve \
+         -var="github_repo=${gh repo you want to use ex https://github.com/aws/amazon-cloudwatch-agent.git}" \
+         -var="github_sha=${commit sha you want to use ex fb9229b9eaabb42461a4c049d235567f9c0439f8}" \
+         -var='vpc_security_group_ids=["${name of your security group}"]' \
+         -var="key_name=${name of key pair your created}" \
+         -var="s3_bucket=${name of your s3 bucket created}" \
+         -var="iam_instance_profile=${name of your iam role created}" \
+         -var="ssh_key=${your key that you downloaded}"
+         ```
```
Comment (adam-mateen) [inline integration/terraform/ec2/linux/README.md:51]: You could add snippets of the expected output, hopefully just a few important lines, not 100s of lines.

**Pull request merged**